components/org.wso2.carbon.identity.conditional.auth.functions.common/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/common/utils/CommonUtils.java

```diff
@@ -18,14 +18,20 @@
 
 package org.wso2.carbon.identity.conditional.auth.functions.common.utils;
 
+import org.apache.commons.lang.StringUtils;
 import org.wso2.carbon.identity.application.common.model.Property;
 import org.wso2.carbon.identity.conditional.auth.functions.common.internal.FunctionsDataHolder;
 import org.wso2.carbon.identity.event.IdentityEventException;
 import org.wso2.carbon.identity.governance.IdentityGovernanceException;
 import org.wso2.carbon.identity.governance.IdentityGovernanceService;
+import org.wso2.carbon.identity.secret.mgt.core.exception.SecretManagementException;
+import org.wso2.carbon.identity.secret.mgt.core.model.ResolvedSecret;
 
 public class CommonUtils {
 
+    private static final String SECRET_TYPE_CALL_CHOREO = "ADAPTIVE_AUTH_CALL_CHOREO";
+    private static final String SECRET_PREFIX = "secrets.";
```

Comment:
Can we address this "secrets" as a object rather than having a string comparison

Reply:
Here we check whether the provided secret alias has the prefix or not. The secret resolving part will happen after this. If the provided value starts with the prefix, then we check the secrets object to resolve the secret.

Reply:
Will provide the improvement in the next phase.

Reply:
shall we move out this code till then